### PR TITLE
fix: Do not retry regional when request is not GET

### PR DESCRIPTION
**Pull request by jwtty**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Only check response body length when request is GET.
#### Which issue(s) this PR fixes:

related #2296 

#### Does this PR introduce a user-facing change?

```release-note
Only check response body length when request is GET. Avoids unnecessary retry when deleting an LB.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by MartinForReal** [comment]
send function accept decorator as parameters and maybe we can move DoHackRegionalRetryDecorator to GetResource function ?

**Comment by feiskyer** [comment]
> send function accept decorator as parameters and maybe we can move DoHackRegionalRetryDecorator to GetResource function ?

yes, this is another way to achieve the same goal. Which approach do you prefer?

File: pkg/azureclients/armclient/azure_armclient.go [hunk 1]
```diff
@@ -362,7 +353,7 @@ func (c *Client) GetResource(ctx context.Context, resourceID string, decorators
 		return nil, retry.NewError(false, err)
 	}
 
-	return c.Send(ctx, request)
```
Comment (MartinForReal) [inline pkg/azureclients/armclient/azure_armclient.go:365]: c.send(ctx,request,doHackRegionalRetryForGET)?

Comment (feiskyer) [reply pkg/azureclients/armclient/azure_armclient.go:365]: +1. prefer keeping RegionalRetry still as a decorator

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-cloud-provider-azure-e2e-ccm-vmss-capz
--


<br class="Apple-interchange-newline">
</body>
</html>

**Comment by MartinForReal** [comment]
LGTM

**Pull request merged**